src/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -63,5 +63,42 @@ public void Pop()
             Debug.Assert(_index > 0);
             Current = _previous[--_index];
         }
+
+        // Return a property path in the form of: [FullNameOfType].FirstProperty.SecondProperty.LastProperty
+        public string PropertyPath
+        {
+            get
+            {
+                StringBuilder path = new StringBuilder();
+
+                if (_previous == null || _index == 0)
+                {
+                    path.Append($"[{Current.JsonClassInfo.Type.FullName}]");
+                }
+                else
+                {
+                    path.Append($"[{_previous[0].JsonClassInfo.Type.FullName}]");
+
+                    for (int i = 0; i < _index; i++)
+                    {
+                        path.Append(GetPropertyName(_previous[i]));
+                    }
+                }
+
+                path.Append(GetPropertyName(Current));
```

Comment:
Is `Current == _previous[_index - 1]`? If so, are we appending the last property twice?

Reply:
`Current` is not stored in `_previous`. `_previous` only has non-current state.